File: MovieCorner.Client/src/app/models/Movie.ts
/**
 * Movie list DTO
 * @export
 * @class MovieListDTO
 */
export class MovieListDTO {
  id: number;
  title: string;
  year: number;
  creater: string;
  description: string;
}

/**
 * Movie DTO
 * @export
 * @class MovieDTO
 */
export class MovieDTO {
  id: number;
  title: string;
  description: string;
  year: number;
}

/**
 * Own Movie DTO
 * @export
 * @class OwnMovieDTO
 */
export class OwnMovieDTO {
  id: number;
  title: string;
  description: string;
  year: number;
  seen: boolean;
}

File: ManagerAPI.Models/Entities/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace ManagerAPI.Models.Entities
{
    public class Movie {
        public int Id { get; set; }

        [Required]
        [MaxLength (150)]
        public string Title { get; set; }

        [MaxLength (999)]
        public string Description { get; set; }

        [Required]
        public int Year { get; set; }

        public string CreaterId { get; set; }
        public virtual User Creater { get; set; }
    }
}
